### DISCUSSION TITLE
Multiple domain static site generation (SSG) by appropriating i18n locales

### DISCUSSION THREAD

**Discussion by enmielado** [vercel/next.js#19970] [Help]
This is a sort of _tell me why this is a bad idea_ post.

With the existing i18n feature it’s possible to render & serve statically generated page variations based on the domain. Since (AFAIK) this is essentially just a mapping of domains to keys and providing those keys to getStaticProps, would there be any issues with using the i18n feature to build a multi-domain site that does essentially the same thing — generates page variations based on the domain?

In this scenario, essentially a locale is a site.

```
// next.config.js
module.exports = {
    i18n: {
        localeDetection: false,
        locales: [‘site-a’, ‘site-b’],
        defaultLocale: 'site-a’,

        domains: [
            {
                domain: 'site-a.example.com',
                defaultLocale: 'site-a’,
            },
            {
                domain: 'site-b.example.com',
                defaultLocale: 'site-b’,
            },
        ],
    },
}
```

```
// pages/post/[slug].js
export async function getStaticProps({ params, locale }) {

  const data = getSiteSpecificPageContent(params, locale)

  …

}
```

I have tested this, and it works just fine.

Potential cons:
- The HTML lang attribute is set to the site/locale key (`<html lang=“site-a”>`), but this can be overcome by setting it in a custom _document.js.
- Some future i18n feature breaks something
- Code confusion (the code is doing something other than what it may appear to do)
- Something else I haven’t anticipated ??

Thanks for your thoughts.

**Comment by tomsoderlund** [comment]
@enmielado here’s an implementation & demo of this concept:

https://github.com/tomsoderlund/nextjs-multi-domain-locale
